src/sagemaker/jumpstart/curated_hub/parsers.py

```diff
@@ -0,0 +1,422 @@
+    specs["inference_volume_size"] = hub_model_document.inference_volume_size
+    specs[
+        "inference_enable_network_isolation"
+    ] = hub_model_document.inference_enable_network_isolation
+    specs["resource_name_base"] = hub_model_document.resource_name_base
+
+    specs["hosting_eula_key"] = None
+    if hub_model_document.hosting_eula_uri is not None:
+        hosting_eula_bucket, hosting_eula_key = parse_s3_url(hub_model_document.hosting_eula_uri)
+        specs["hosting_eula_key"] = hosting_eula_key
+
+    if hub_model_document.hosting_model_package_arn:
+        specs["hosting_model_package_arns"] = {region: hub_model_document.hosting_model_package_arn}
+
+    specs["hosting_use_script_uri"] = hub_model_document.hosting_use_script_uri
+
+    specs["hosting_instance_type_variants"] = hub_model_document.hosting_instance_type_variants
+
+    if specs["training_supported"]:
+        specs["training_ecr_uri"] = hub_model_document.training_ecr_uri
+        training_artifact_bucket, training_artifact_key = parse_s3_url(
+            hub_model_document.training_artifact_uri
+        )
+        specs["training_artifact_key"] = training_artifact_key
+        training_script_bucket, training_script_key = parse_s3_url(
+            hub_model_document.training_script_uri
+        )
+        specs["training_script_key"] = training_script_key
+
+        specs["hyperparameters"] = hub_model_document.hyperparameters
+        specs["training_volume_size"] = hub_model_document.training_volume_size
+        specs[
+            "training_enable_network_isolation"
+        ] = hub_model_document.training_enable_network_isolation
+        if hub_model_document.training_model_package_artifact_uri:
+            specs["training_model_package_artifact_uris"] = {
+                region: hub_model_document.training_model_package_artifact_uri
+            }
+        specs[
+            "training_instance_type_variants"
+        ] = hub_model_document.training_instance_type_variants
+    return JumpStartModelSpecs(_to_json(specs), is_hub_content=True)
+
+
+def make_hub_model_document_from_specs(
+    model_specs: JumpStartModelSpecs,
+    studio_specs: Dict[str, Any],
+    hub_content_dependencies: List[HubContentDependency],
+    region: str,
+) -> HubModelDocument:
+    """Sets fields in HubModelDocument based on model specs, studio specs,
+    and hub content dependencies.
+    """
+    document = {}
+    document["Url"] = model_specs.url
+    document["MinSdkVersion"] = model_specs.min_sdk_version
+    document["HostingEcrUri"] = image_uris.retrieve(
+        model_id=model_specs.model_id,
+        model_version=model_specs.version,
+        framework=model_specs.hosting_ecr_specs.framework,
+        instance_type=model_specs.default_inference_instance_type,
+        image_scope=JumpStartScriptScope.INFERENCE,
+        region=region,
+    )
+    document["GatedBucket"] = model_specs.gated_bucket
+    content_bucket = (
```

Comment:
The content bucket is not the same for all model resources when a model is gated. Only artifacts are stored in the gated bucket right now, everything else is stored in the public bucket.